### PR TITLE
feat: add a time limit to heuristic optimization

### PR DESCRIPTION
**Pull request by joaocardoso**
Also added fix documentation and unit tests.

### CODE REVIEW

File: cameo/strain_design/heuristic/evolutionary/optimization.py [hunk 1]
```diff
@@ -205,7 +229,21 @@ def run(self, evaluator=None, generator=None, view=config.default_view, maximize
 
         for observer in self.observers:
             observer.reset()
+
         t = time.time()
+
+        if max_time is not None:
+            terminator = self.heuristic_method.terminator
```
Comment (phantomas1234) [inline cameo/strain_design/heuristic/evolutionary/optimization.py]: What if `self.heuristic_method.terminator` is already `inspyred.ec.terminators.time_termination`?

File: cameo/api/designer.py [hunk 2]
```diff
@@ -74,9 +74,11 @@ def __call__(self, strategy):
 class _OptGeneRunner(_OptimizationRunner):
     def __call__(self, strategy):
         max_evaluations = 15000
```
Comment (phantomas1234) [inline cameo/api/designer.py]: All of this should not be hard-coded, but let's refactor this later.

File: cameo/strain_design/heuristic/evolutionary/optimization.py (hunk 1 above)
Comment (joaocardoso) [inline cameo/strain_design/heuristic/evolutionary/optimization.py]: then they will both trigger termination with the same max_time so no problem.

File: cameo/api/designer.py (hunk 2 above)
Comment (joaocardoso) [inline cameo/api/designer.py]: I agree, but it needs a general configuration strategy with default values that can be easily defined without using 50 keyword arguments.

**Pull request merged**